### PR TITLE
Added initial setup for level 2

### LINKED ISSUE #1225: Develop Level 2 of the game
## Description
As a developer,
I need add level to the game,
so that I can make the app more involving.

## Mocks
Not applicable

### Update
- [ ] The map should reset to initial state with only first scenario unlocked
- [ ] The conversations should be generic
- [ ] Proper mechanism to launch the mini games
- [ ] The avatar should be the high school version

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
30 hours

### PR DESCRIPTION
**Pull request by Rimjhim28**
### Description
In this PR I have added initial code for Level 2 that is implemented mainly through GameLevel2Activity. java, ScenarioOverLevel2Activity.java and MapLevel2Activity.java.
The changes are:
- After the completion of Level 1 of the game, Level 2 can be launched by clicking the MAP button on game over screen. Right now, I have not done any changes on game over screen since this is an UI change and requires views from other members.
- Map is reset to initial state. Only first scenario is unlocked and rest are locked.
- I have used dummy data for the questions, answers and scenario csv files that populate the game activity
- I have attached the two developed mini games. Kill the Virus is attached to first scenario for Level 2. Since Memory Match is used to make the user familiar with the tiles used in Vocab Match I replaced VocabMatch with MemoryMatch ( launches after completing Hospital scenario of Level 1 ). VocabMatch launches after Hospital scenario of Level 2.

Since this is just a initial setup, the code changes makes sure that the data is fetched correctly and all the mini games launch as expected. I will attach session managers, change the avatar and do ther code modiications in next PR.

Fixes #1225 

### Type of Change

- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Next Level of the game can be launched by pressing MAP on GameOver screen that appears after completion of level 1.

### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: PowerUp/app/src/main/java/powerup/systers/com/ScenarioOverLevel2Activity.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+    public ScenarioOverLevel2Activity(Context context) {
+        this.context_scenario = context;
+        sharedPreferences_scenario = context.getSharedPreferences(PREF_NAME_SCENARIO, PRIVATE_MODE_SCENARIO);
+        editor_scenario = sharedPreferences_scenario.edit();
+    }
+
+    @TargetApi(Build.VERSION_CODES.JELLY_BEAN)
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```
[the remark this answers is not in the document]
Comment (Rimjhim28) [reply PowerUp/app/src/main/java/powerup/systers/com/ScenarioOverLevel2Activity.java]: I don't know how to resolve this!

File: PowerUp/app/src/main/java/powerup/systers/com/GameLevel2Activity.java [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+import android.text.method.ScrollingMovementMethod;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import java.util.ArrayList;
```
[the remark this answers is not in the document]
Comment (Rimjhim28) [reply PowerUp/app/src/main/java/powerup/systers/com/GameLevel2Activity.java]: The variable is actually used. This is an invalid error.

File: PowerUp/app/src/main/java/powerup/systers/com/ScenarioOverLevel2Activity.java [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            }
+        });
+        if (getIntent().getExtras() != null && PowerUpUtils.MAP.equals(getIntent().getExtras().getString(PowerUpUtils.SOURCE))) {
+            continueButton.setVisibility(View.GONE);
+            continueButton.setOnClickListener(null);
+        }
+
+
+        replayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
```
Comment (thisum) [inline PowerUp/app/src/main/java/powerup/systers/com/ScenarioOverLevel2Activity.java]: use butterknife click binders

File: PowerUp/app/src/main/java/powerup/systers/com/ScenarioOverLevel2Activity.java [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        ButterKnife.bind(this);
+        scene = getmDbHandler().getScenario();
+        final Scenario prevScene = getmDbHandler().getScenarioFromID(SessionHistory.prevSessionID); //Fetching Scenario
+        scenarioActivityDone = 1;
+        //If not launched from map then only dialogMaker() is called
+        if (!new ScenarioOverLevel2Activity(this).isActivityOpened() && (!(getIntent().getExtras() != null && PowerUpUtils.MAP.equals(getIntent().getExtras().getString(PowerUpUtils.SOURCE))))) {
+            dialogMaker();
+        }
+        mapButton.setOnClickListener(new View.OnClickListener() {
+            @Override
```
Comment (thisum) [inline PowerUp/app/src/main/java/powerup/systers/com/ScenarioOverLevel2Activity.java]: use butterknife onclick binders

File: PowerUp/app/src/main/java/powerup/systers/com/ScenarioOverLevel2Activity.java [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+        //Initializing and setting Text for currentScenarioName
+        if (getIntent().getExtras() != null && PowerUpUtils.MAP.equals(getIntent().getExtras().getString(PowerUpUtils.SOURCE)) && getIntent().getStringExtra(PowerUpUtils.SCENARIO_NAME) != null)
+            currentScenarioName.setText(getResources().getString(R.string.current_scenario_name, getIntent().getStringExtra(PowerUpUtils.SCENARIO_NAME)));
+        else
+            currentScenarioName.setText(getResources().getString(R.string.current_scenario_name, prevScene.getScenarioName()));
+
+        karmaPoints.setText(String.valueOf(SessionHistory.totalPoints));
+        continueButton.setOnClickListener(new View.OnClickListener() {
+            @Override
```
Comment (thisum) [inline PowerUp/app/src/main/java/powerup/systers/com/ScenarioOverLevel2Activity.java]: use butterknife onclick binders

File: PowerUp/app/src/main/java/powerup/systers/com/GameLevel2Activity.java [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+            error.printStackTrace();
+        }
+
+        // Update Scene
+        updateScenario(0);
+        updateQA();
+        //Scene is Replayed
+        if (scene.getReplayed() == 1) {
+            goToMap.setAlpha((float) 1.0);
+            goToMap.setOnClickListener(new View.OnClickListener() {
```
Comment (thisum) [inline PowerUp/app/src/main/java/powerup/systers/com/GameLevel2Activity.java]: use butterknife onclick binders

File: PowerUp/app/src/main/java/powerup/systers/com/GameLevel2Activity.java [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+        }
+        if (scene != null)
+            prevScene = getmDbHandler().getScenarioFromID(scene.getId());
+        scene = getmDbHandler().getScenario();
+        // Play the scenario first time
+        if (scene.getReplayed() == 0) {
+            // goToMap Mechanics
+            goToMap.setAlpha((float) 1.0);
+            goToMap.setOnClickListener(new View.OnClickListener() {
+                @Override
```
Comment (thisum) [inline PowerUp/app/src/main/java/powerup/systers/com/GameLevel2Activity.java]: use butterknife onclick binders

**Comment by thisum** [comment]
@Rimjhim28 I just saw this merge conflict. Pls resolve that as well

**Comment by Rimjhim28** [comment]
@thisum , @debanjanah has already resolved the conflicts.

**Pull request merged**